precode.go

```diff
@@ -40,13 +40,70 @@ var tasks = map[string]Task{
 }
 
 // Ниже напишите обработчики для каждого эндпоинта
-// ...
+func getTasks(w http.ResponseWriter, r *http.Request) {
+    resp, err := json.MarshalIndent(tasks, "", "   ")
+    if err != nil {
+        http.Error(w, err.Error(), http.StatusInternalServerError)
+        return
+    }
+	w.Header().Set("Content-Type", "application/json")
+    	w.WriteHeader(http.StatusOK)
+    	w.Write(resp)
+}	
+
+func postTask(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	var task Task
+	if err := json.NewDecoder(r.Body).Decode(&task); err != nil {
+		http.Error(w, "400 Bad Request", http.StatusBadRequest)
+		return
+	}
+	if _, ok := tasks[task.ID]; ok {
+		http.Error(w, "400 Bad Request", http.StatusBadRequest)
```

Comment:
Лучше как-то содержательное сообщение писать
```suggestion
		http.Error(w, "task with this id is already exists", http.StatusBadRequest)
```